### PR TITLE
DROOLS-733: kie-server-client refctored tests with wiremock to execut…

### PR DESCRIPTION
**Pull request by mswiderski**
…e them without a need to have running kie server

### CODE REVIEW

**Comment by mswiderski** [comment]
following our discussion on improving kid-server-client tests in osgi/karaf, here is a PR that does that using suggested WireMock. @psiroky if you could take a look whenever you have some time.

File: drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/util/PaxExamWithWireMock.java [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+import com.github.tomakehurst.wiremock.WireMockServer;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runners.model.InitializationError;
+import org.ops4j.pax.exam.junit.PaxExam;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.*;
+
+public class PaxExamWithWireMock extends PaxExam {
+
```
Comment (psiroky) [inline drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/util/PaxExamWithWireMock.java]: Maybe we could use `PaxExamParameterized` it basically combines PaxExam with default jUnit's Parametrized runner. We could hook the wiremock as the tests are usually the same for XML and JSON?

**Comment by psiroky** [comment]
+1. After talking to @mswiderski over IRC, it is not easily possible to use the `PaxExamParametrized` (will be explaining why as a comment in the test class).

**Comment by mswiderski** [comment]
merged into master and 6.3.x

**Pull request closed without merging**